src/Formatter/FormatterManager.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\restful\Formatter\FormatterManager
+ */
+
+namespace Drupal\restful\Formatter;
+
+use Drupal\restful\Plugin\formatter\FormatterInterface;
+use Drupal\restful\Plugin\FormatterPluginManager;
+use Drupal\Component\Plugin\Exception\PluginNotFoundException;
+
+class FormatterManager implements FormatterManagerInterface {
+
+  /**
+   * The rate limit plugins.
+   *
+   * @var FormatterPluginCollection
+   */
+  protected $plugins;
+
+  /**
+   * Constructs FormatterManager.
+   *
+   * @param \RestfulBase $resource
+   *   TODO: Remove this coupling.
+   *   The resource.
+   */
+  public function __construct($resource = NULL) {
+    $this->resource = $resource;
+    $manager = FormatterPluginManager::create();
+    $options = array();
+    foreach ($manager->getDefinitions() as $plugin_id => $plugin_definition) {
+      // Since there is only one instance per plugin_id use the plugin_id as
+      //instance_id.
+      // Add the restful resource to the plugin options.
+      // TODO: Remove this coupling.
```

Comment:
Ideally the formatter should not need to be composed with the resource. We are creating circular dependencies (since the resource is composed with the formatter via the formatter manager), and thus coupling those classes very tightly.

Not sure how to deal with this as of now, that's why I put a TODO.